### PR TITLE
TST: un-xfail NaT test on npdev

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/scalar/test_nat.py [hunk 1]
```diff
@@ -524,27 +525,24 @@ def test_to_numpy_alias():
...
+        Timestamp(0).to_datetime64(),
         Timestamp(0).tz_localize("UTC"),
         NaT,
     ],
 )
-def test_nat_comparisons(compare_operators_no_eq_ne, other):
+def test_nat_comparisons(compare_operators_no_eq_ne, other, request):
     # GH 26039
     opname = compare_operators_no_eq_ne
+    if isinstance(other, (np.datetime64, np.timedleta64)) and (
```
Comment (mroeschke) [inline pandas/tests/scalar/test_nat.py:539]:
```suggestion
    if isinstance(other, (np.datetime64, np.timedelta64)) and (
```

File: pandas/tests/scalar/test_nat.py [hunk 2]
```diff
@@ -524,27 +525,24 @@ def test_to_numpy_alias():
...
         Timestamp(0).tz_localize("UTC"),
         NaT,
     ],
 )
-def test_nat_comparisons(compare_operators_no_eq_ne, other):
+def test_nat_comparisons(compare_operators_no_eq_ne, other, request):
     # GH 26039
     opname = compare_operators_no_eq_ne
+    if isinstance(other, (np.datetime64, np.timedelta64)) and (
+        opname in ["__eq__", "__ne__"] or not is_numpy_dev
```
Comment (jbrockmendel) [inline pandas/tests/scalar/test_nat.py:540]: looks like the eq/ne checks here are unnecessary.  ill clean it up in a follow-up so we can get the CI back to green quickly.

**Review by mroeschke — APPROVED** [review]
LGTM merge whoever sees greenish first

**Comment by jbrockmendel** [comment]
npdev is green, the failing test is an unrelated file leak. merging.

**Pull request merged**